File: Backend/Data/ViewModels/Response/UserVM.cs
namespace Backend.Data.ViewModels
{
    /// <summary>    
    /// View Model class for user's account view reqeust and edit user's account response
    /// </summary>
    public class UserVM
    {
        public string UserName { get; set; }
        public string Email { get; set; }
        public string Name { get; set; }
        public DateTime BirthDate { get; set; }
        public Boolean PrivateProfile { get; set; }
        public string Description { get; set; }
        public string ProfilePhoto { get; set; }

        /// <summary>
        /// Class' constructor method
        /// </summary>
        /// <param name="userName">User's unique identifier</param>
        /// <param name="email">User's email</param>
        /// <param name="name">User's name</param>
        /// <param name="birthDate">User's birth date</param>
        /// <param name="privateProfile">Whether or not user's profile is private</param>
        /// <param name="description">User's profile description</param>
        /// <param name="profilePhoto">User's profile photo </param>
        public UserVM(string userName, string email, string name, DateTime birthDate, Boolean privateProfile,
         string description, string profilePhoto)
        {
            this.UserName = userName;
            this.Email = email;
            this.Name = name;
            this.BirthDate = birthDate;
            this.PrivateProfile = privateProfile;
            this.Description = description;
            this.ProfilePhoto = profilePhoto;
        }


    }
}

File: Frontend/src/app/viewModels/responses/UserVM.ts
import { toJSONString } from "../interfaces/toJSONString";

/**
 * View Model that represents an User with all it's properties
 */
export class UserVM implements toJSONString{

    /**
     * Construct's an instance of {@link UserVM}
     * 
     * @param userName user's user name
     * @param email user's email
     * @param name user's name
     * @param birthDate user's birth date
     * @param privateProfile Specifies if the user has a private profile
     * @param description user's description
     * @param profilePhoto user's profile photo url
     */
    constructor(
        public userName: string,
        public email: string,
        public name: string,
        public birthDate: string,
        public privateProfile: Boolean,
        public description: string,
        public profilePhoto: string,
    ) { }

    /**
     * Converts the class into a string representation of a json object 
     * 
     * @returns string representation of the object
     */
    toJSONString(): string {
        return JSON.stringify(this)
    }

}